**Assignment: Restaurant Branch Performance EDA (Day 13)**

Using the provided Restaurant Branch Performance Dataset (found in the "Study Material" tab in LMS), perform an Exploratory Data Analysis (EDA) using Pandas and the concepts covered in Day 13. Explore the dataset through summary statistics, descriptive analysis, distribution analysis, and correlation analysis to understand the relationships and patterns within the data. Investigate important factors such as customers, revenue, profit, marketing spend, staff count, delivery time, and customer ratings, compare performance across branches, regions, and store types, and use an appropriate correlation matrix to identify meaningful relationships between numerical variables. Conclude your analysis with 5 to 8 meaningful, data-driven observations explaining what you discovered from the dataset, and submit your Jupyter/Google Colab notebook containing the complete EDA.

In [5]:
import pandas as pd

In [6]:
data = pd.read_csv("Day13_Restaurant_Branch_Performance_Dataset.csv")
data

,Record_ID,Date,Day,Month,Branch,Region,Store_Type,Staff_Count,Marketing_Spend,Customers,...,Revenue,Food_Cost,Operating_Cost,Profit,Avg_Delivery_Min,Customer_Rating,Temperature_C,Humidity_Percent,Weather,Promotion
0,ST0001,2026-06-06,Saturday,June,Jaipur,North,Express,7,4977.25,172,...,66766.77,27954.29,46554.90,20211.87,29.0,4.44,20.8,70.3,Clear,NaN
1,ST0002,2026-02-10,Tuesday,February,Pune,West,Premium,14,8998.30,285,...,157230.89,63816.59,96813.95,60416.94,27.9,4.56,31.3,70.7,Clear,Loyalty Offer
2,ST0003,2026-03-07,Saturday,March,Delhi,North,Premium,16,10034.97,306,...,155811.96,54874.10,89888.91,65923.05,30.2,4.36,21.4,61.5,Clear,NaN
3,ST0004,2026-05-11,Monday,May,Mumbai,West,Standard,9,5095.84,198,...,80574.05,31429.44,53531.82,27042.23,21.7,4.65,22.9,40.4,Clear,NaN
4,ST0005,2026-01-14,Wednesday,January,Kochi,South,Express,9,6681.82,148,...,56579.89,21759.51,44376.25,12203.64,26.1,4.50,29.0,44.3,Clear,Combo Deal
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
345,ST0346,2026-02-08,Sunday,February,Mumbai,West,Express,10,7389.17,202,...,56852.25,18863.88,43733.92,13118.33,27.2,4.63,28.9,64.6,Rain,NaN
346,ST0347,2026-06-05,Friday,June,Mumbai,West,Premium,12,8398.13,274,...,138198.35,50914.04,75370.62,62827.73,35.4,4.32,22.7,57.9,Rain,NaN
347,ST0348,2026-03-13,Friday,March,Bengaluru,South,Standard,16,2876.93,196,...,62974.97,23946.23,51377.97,11597.00,22.5,4.51,32.3,63.4,Clear,Combo Deal
348,ST0349,2026-03-10,Tuesday,March,Jaipur,North,Standard,9,4731.74,200,...,78451.05,26050.73,47400.29,31050.76,28.3,4.51,24.7,58.6,Clear,Loyalty Offer


In [7]:
data.shape

(350, 22)

In [8]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 350 entries, 0 to 349
Data columns (total 22 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Record_ID         350 non-null    object 
 1   Date              350 non-null    object 
 2   Day               350 non-null    object 
 3   Month             350 non-null    object 
 4   Branch            350 non-null    object 
 5   Region            350 non-null    object 
 6   Store_Type        350 non-null    object 
 7   Staff_Count       350 non-null    int64  
 8   Marketing_Spend   350 non-null    float64
 9   Customers         350 non-null    int64  
 10  Orders            350 non-null    int64  
 11  Average_Bill      350 non-null    float64
 12  Revenue           350 non-null    float64
 13  Food_Cost         350 non-null    float64
 14  Operating_Cost    350 non-null    float64
 15  Profit            350 non-null    float64
 16  Avg_Delivery_Min  350 non-null    float64
 1

In [9]:
data.describe()

,Staff_Count,Marketing_Spend,Customers,Orders,Average_Bill,Revenue,Food_Cost,Operating_Cost,Profit,Avg_Delivery_Min,Customer_Rating,Temperature_C,Humidity_Percent
count,350.000000,350.000000,350.000000,350.000000,350.000000,350.000000,350.000000,350.000000,350.000000,350.000000,350.000000,350.000000,350.000000
mean,11.957143,5516.347743,217.377143,187.054286,388.906371,86831.799657,31807.545114,56621.444486,30210.355171,26.098857,4.469314,27.740000,61.324571
std,4.338261,2050.906967,56.722524,50.558371,82.488196,36644.453901,13896.888942,18467.842215,19508.592192,4.388125,0.198258,4.903148,13.757322
min,5.000000,725.350000,77.000000,71.000000,180.000000,20540.760000,7262.010000,20782.380000,-6924.680000,13.900000,3.760000,14.500000,25.000000
25%,9.000000,4339.727500,180.250000,155.000000,337.315000,61261.717500,22326.662500,43797.670000,16915.492500,22.800000,4.320000,24.400000,51.325000
50%,11.000000,5393.370000,213.000000,182.500000,383.485000,79766.880000,28861.325000,53723.025000,26873.870000,26.100000,4.470000,27.800000,61.650000
75%,15.000000,6748.502500,255.750000,218.750000,441.765000,102976.185000,38370.370000,63961.227500,39823.742500,29.075000,4.600000,31.000000,70.375000
max,24.000000,11784.960000,360.000000,334.000000,623.280000,194018.920000,75946.590000,111489.140000,89096.190000,38.300000,4.960000,43.500000,95.000000


In [10]:
data.isnull().sum()

Record_ID             0
Date                  0
Day                   0
Month                 0
Branch                0
Region                0
Store_Type            0
Staff_Count           0
Marketing_Spend       0
Customers             0
Orders                0
Average_Bill          0
Revenue               0
Food_Cost             0
Operating_Cost        0
Profit                0
Avg_Delivery_Min      0
Customer_Rating       0
Temperature_C         0
Humidity_Percent      0
Weather               0
Promotion           184
dtype: int64

In [11]:
data["Promotion"] = data["Promotion"].fillna("No Promotion")
data.head()

,Record_ID,Date,Day,Month,Branch,Region,Store_Type,Staff_Count,Marketing_Spend,Customers,...,Revenue,Food_Cost,Operating_Cost,Profit,Avg_Delivery_Min,Customer_Rating,Temperature_C,Humidity_Percent,Weather,Promotion
0,ST0001,2026-06-06,Saturday,June,Jaipur,North,Express,7,4977.25,172,...,66766.77,27954.29,46554.90,20211.87,29.0,4.44,20.8,70.3,Clear,No Promotion
1,ST0002,2026-02-10,Tuesday,February,Pune,West,Premium,14,8998.30,285,...,157230.89,63816.59,96813.95,60416.94,27.9,4.56,31.3,70.7,Clear,Loyalty Offer
2,ST0003,2026-03-07,Saturday,March,Delhi,North,Premium,16,10034.97,306,...,155811.96,54874.10,89888.91,65923.05,30.2,4.36,21.4,61.5,Clear,No Promotion
3,ST0004,2026-05-11,Monday,May,Mumbai,West,Standard,9,5095.84,198,...,80574.05,31429.44,53531.82,27042.23,21.7,4.65,22.9,40.4,Clear,No Promotion
4,ST0005,2026-01-14,Wednesday,January,Kochi,South,Express,9,6681.82,148,...,56579.89,21759.51,44376.25,12203.64,26.1,4.50,29.0,44.3,Clear,Combo Deal


In [12]:
data.isnull().sum().sum()

np.int64(0)

In [13]:
data.nunique()

Record_ID           350
Date                161
Day                   7
Month                 6
Branch                8
Region                3
Store_Type            3
Staff_Count          20
Marketing_Spend     350
Customers           178
Orders              170
Average_Bill        348
Revenue             350
Food_Cost           350
Operating_Cost      350
Profit              350
Avg_Delivery_Min    154
Customer_Rating      84
Temperature_C       166
Humidity_Percent    253
Weather               4
Promotion             4
dtype: int64

In [14]:
data.duplicated().sum()

np.int64(0)

In [16]:
data.mean

<bound method DataFrame.mean of     Record_ID        Date        Day     Month     Branch Region Store_Type  \
0      ST0001  2026-06-06   Saturday      June     Jaipur  North    Express   
1      ST0002  2026-02-10    Tuesday  February       Pune   West    Premium   
2      ST0003  2026-03-07   Saturday     March      Delhi  North    Premium   
3      ST0004  2026-05-11     Monday       May     Mumbai   West   Standard   
4      ST0005  2026-01-14  Wednesday   January      Kochi  South    Express   
..        ...         ...        ...       ...        ...    ...        ...   
345    ST0346  2026-02-08     Sunday  February     Mumbai   West    Express   
346    ST0347  2026-06-05     Friday      June     Mumbai   West    Premium   
347    ST0348  2026-03-13     Friday     March  Bengaluru  South   Standard   
348    ST0349  2026-03-10    Tuesday     March     Jaipur  North   Standard   
349    ST0350  2026-02-03    Tuesday  February       Pune   West   Standard   

     Staff_Count  M

In [17]:
data.median

<bound method DataFrame.median of     Record_ID        Date        Day     Month     Branch Region Store_Type  \
0      ST0001  2026-06-06   Saturday      June     Jaipur  North    Express   
1      ST0002  2026-02-10    Tuesday  February       Pune   West    Premium   
2      ST0003  2026-03-07   Saturday     March      Delhi  North    Premium   
3      ST0004  2026-05-11     Monday       May     Mumbai   West   Standard   
4      ST0005  2026-01-14  Wednesday   January      Kochi  South    Express   
..        ...         ...        ...       ...        ...    ...        ...   
345    ST0346  2026-02-08     Sunday  February     Mumbai   West    Express   
346    ST0347  2026-06-05     Friday      June     Mumbai   West    Premium   
347    ST0348  2026-03-13     Friday     March  Bengaluru  South   Standard   
348    ST0349  2026-03-10    Tuesday     March     Jaipur  North   Standard   
349    ST0350  2026-02-03    Tuesday  February       Pune   West   Standard   

     Staff_Count 

In [18]:
data.mode

<bound method DataFrame.mode of     Record_ID        Date        Day     Month     Branch Region Store_Type  \
0      ST0001  2026-06-06   Saturday      June     Jaipur  North    Express   
1      ST0002  2026-02-10    Tuesday  February       Pune   West    Premium   
2      ST0003  2026-03-07   Saturday     March      Delhi  North    Premium   
3      ST0004  2026-05-11     Monday       May     Mumbai   West   Standard   
4      ST0005  2026-01-14  Wednesday   January      Kochi  South    Express   
..        ...         ...        ...       ...        ...    ...        ...   
345    ST0346  2026-02-08     Sunday  February     Mumbai   West    Express   
346    ST0347  2026-06-05     Friday      June     Mumbai   West    Premium   
347    ST0348  2026-03-13     Friday     March  Bengaluru  South   Standard   
348    ST0349  2026-03-10    Tuesday     March     Jaipur  North   Standard   
349    ST0350  2026-02-03    Tuesday  February       Pune   West   Standard   

     Staff_Count  M

In [19]:
data['Revenue'].skew()

np.float64(0.8620269730776782)

In [20]:
data['Region'].value_counts()

Region
South    137
North    132
West      81
Name: count, dtype: int64

In [21]:
data["Promotion"].value_counts()

Promotion
No Promotion     184
Weekend Offer     74
Combo Deal        59
Loyalty Offer     33
Name: count, dtype: int64

In [22]:
data.groupby('Store_Type')[['Revenue', 'Profit', 'Customers']].mean()


,Revenue,Profit,Customers
Store_Type,,,
Express,52327.164804,12742.137941,161.294118
Premium,152867.313269,62890.235385,298.730769
Standard,87268.667245,30630.785714,224.979592


In [23]:
print(data.groupby('Region')[['Revenue', 'Profit', 'Customers']].mean())

             Revenue        Profit   Customers
Region                                        
North   86946.939470  30625.960833  217.651515
South   88320.217518  30690.568102  218.934307
West    84126.716914  28720.859877  214.296296


In [24]:
data.groupby('Branch')[['Revenue', 'Profit', 'Customer_Rating']].mean()

,Revenue,Profit,Customer_Rating
Branch,,,
Bengaluru,90732.118696,31890.914783,4.465652
Delhi,87573.869375,31385.295313,4.485625
Hyderabad,88804.899535,31402.976977,4.469535
Jaipur,86746.158333,30469.703333,4.468958
Kochi,85574.617917,28902.036250,4.419792
Mumbai,80779.248864,26580.988409,4.442500
Pune,88107.489730,31265.571892,4.540811
Srinagar,86746.472885,30302.915769,4.480192


In [25]:
corr_matrix = data.corr(numeric_only=True)
print("correlation of revenue column :")
print(corr_matrix['Revenue'].sort_values(ascending=False))

correlation of revenue column :
Revenue             1.000000
Food_Cost           0.978581
Profit              0.966831
Operating_Cost      0.962914
Customers           0.876361
Orders              0.857756
Average_Bill        0.843033
Staff_Count         0.695114
Marketing_Spend     0.431136
Avg_Delivery_Min    0.078713
Customer_Rating     0.007942
Temperature_C      -0.026119
Humidity_Percent   -0.091142
Name: Revenue, dtype: float64


**Correlation Matrix Observations:**

- Customers and Orders are almost perfectly related (0.98) — more footfall directly means more orders.
- Revenue is strongly tied to Food_Cost (0.98), Profit (0.97) and Operating_Cost (0.96).
- Customer volume factors (Customers 0.88, Orders 0.86, Average_Bill 0.84) drive revenue strongly.
- Marketing_Spend shows only a moderate link with Revenue (0.43) and Profit (0.35).
- The only meaningful negative relationship is Delivery Time vs Customer Rating (-0.44) — slower delivery hurts ratings.
- Weather factors (Temperature, Humidity) have almost no impact on any business metric.

## Observations & Key Findings

**1. Dataset Overview**
The dataset has 350 records and 22 columns. Only Promotion column was having missing values (184 rows out of 350).
I filled them with "No Promotion" because on those days simply no offer was running. There were no duplicate records in the dataset.

**2. Revenue Distribution**
Average revenue is around ₹86,831 but it ranges from ₹20,540 to ₹1,94,018.
Skewness of revenue is +0.86 which means most of the branch-days earn normal revenue and only few days earn very high revenue.

**3. Some Days Are in Loss**
Profit is positive for most of the records but the minimum profit is -₹6,924.
This means some operating days ended in loss, so not every day is profitable for the branches.

**4. Premium Stores Are The Best Performers**
Premium store type has average revenue of ₹152,867 which is almost 3 times of Express stores (₹52,327).
Their average profit (₹62,890) is nearly 5 times of Express profit (₹12,742). Clearly Premium format is winning.

**5. Region and Branch Comparison**
South region is performing slightly better with average revenue ₹88,320 and West region is a little behind with ₹84,127,
but overall all three regions are very close to each other.
Bengaluru branch has the highest average revenue (₹90,732) and Mumbai is at the bottom (₹80,779) with lowest profit also.
In customer ratings, Pune is the best branch (4.54) and Kochi is the lowest (4.42).

**6. What Drives Revenue**
Correlation matrix shows Revenue is strongly connected with Food_Cost (0.98), Profit (0.97), Operating_Cost (0.96),
Customers (0.88) and Orders (0.86). So basically when more customers come and more orders happen, revenue increases automatically.

**7. Marketing Alone Is Not Enough**
Marketing Spend is showing only moderate correlation with Revenue (0.43) and Profit (0.35).
So spending more money on marketing does not guarantee high revenue.

**8. Service Quality and Weather Effect**
Delivery time and customer rating are having a negative relation (-0.44), which means slow delivery reduces the rating.
Customer rating is having almost zero connection with revenue (0.008), and weather columns (temperature, humidity)
are not affecting any business metric. Revenue depends more on footfall than weather or service scores.